livepeer/stream-tester
# Add E2E Testing for NFT SDK

https://github.com/livepeer/stream-tester/issues/146

1. Refactor `record-tester` to use `go-api-client`

2. After `record-tester` finishes ->
* Wait for Asset to be available
* Run transcode task on that asset
* Wait for transcoding asset to complete
* Run export task on transcoded asset
* Wait for export to complete
* Check that IPFS link is available on export task

Testing locally:
```
~/livepeer/stream-tester % make
~/livepeer/stream-tester % ./build/recordtester -api-token 29b59cfa-3006-48b7-a5fd-fbd07073372e  -file https://storage.googleapis.com/lp_testharness_assets/bbb_sunflower_1080p_30fps_normal_2min.mp4 -test-dur 4m -continuous-test 15s -vod -stream-health true
...
I0517 04:09:47.721283   45384 recordtester_app.go:306] Record testing done, waiting for Asset to be available
I0517 04:09:47.721300   45384 recordtester_app.go:316] Waiting 30s for asset, elapsed=0s
I0517 04:10:18.172198   45384 recordtester_app.go:316] Waiting 30s for asset, elapsed=30s
I0517 04:10:48.666966   45384 recordtester_app.go:316] Waiting 30s for asset, elapsed=1m0s
I0517 04:11:19.198480   45384 recordtester_app.go:316] Waiting 30s for asset, elapsed=1m30s
I0517 04:11:50.226989   45384 recordtester_app.go:348] Asset is available id=56904bbd-46c2-4bc9-9d6a-874442340b22, transcoding taskId=cb594fd5-197c-4769-b1f8-27984623c66d outputAssetId=7b526a4e-4d28-40c7-a224-b8699c4695c9
I0517 04:11:50.227012   45384 recordtester_app.go:353] Waiting 15s for transcode output asset id=7b526a4e-4d28-40c7-a224-b8699c4695c9, elapsed=0s
I0517 04:12:06.143579   45384 recordtester_app.go:384] Transcode output asset id=7b526a4e-4d28-40c7-a224-b8699c4695c9 ready, exporting, taskId=1a51823c-3186-4330-a3a5-121335cacbad
I0517 04:12:06.143610   45384 recordtester_app.go:389] Waiting 5s for asset id=7b526a4e-4d28-40c7-a224-b8699c4695c9 to be exported, elapsed=0s
I0517 04:12:11.562321   45384 recordtester_app.go:403] Export success, task id=1a51823c-3186-4330-a3a5-121335cacbad ipfs link=https://ipfs.livepeer.com/ipfs/QmfPT1LxgpP1achTWECXkghTRqb3j7iifJSLDp4ergqSK2
I0517 04:12:11.562350   45384 recordtester_app.go:409] Done
```

go-api-client PR: https://github.com/livepeer/go-api-client/pull/7
